akka-stream/src/main/scala/akka/stream/impl/Sinks.scala

```diff
@@ -441,3 +445,102 @@ private[akka] final class ReducerState[T, R](val collector: java.util.stream.Col
   def finish(): R = collector.finisher().apply(reduced)
 }
 
+/**
+ * INTERNAL API
+ */
+final private[stream] class LazySink[T, M](sinkFactory: T ⇒ Future[Sink[T, M]], zeroMat: () ⇒ M) extends GraphStageWithMaterializedValue[SinkShape[T], Future[M]] {
```

Comment:
Let's go with the wrapping API, we can add sugar later in a non-breaking way anyway.